src/DataStructures/DiagonalModalOperator.hpp

```diff
@@ -0,0 +1,108 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataVector.hpp"  // IWYU pragma: keep
+#include "DataStructures/VectorImpl.hpp"
+#include "Utilities/PointerVector.hpp"  // IWYU pragma: keep
+#include "Utilities/TMPL.hpp"
+
+class ModalVector;
+
+// IWYU pragma: no_include <blaze/math/expressions/DVecMapExpr.h>
+// IWYU pragma: no_include <blaze/math/typetraits/IsVector.h>
+
+/*!
+ * \ingroup DataStructuresGroup
+ * \brief A class for an element-wise multiplier of modal coefficients.
+ *
+ * \details A `DiagonalModalOperator` holds an array of factors to multiply by
+ * spectral coefficients, and can be either owning (the array is deleted when
+ * the `DiagonalModalOperator` goes out of scope) or non-owning, meaning it just
+ * has a pointer to an array.
+ *
+ * Only basic mathematical operations are supported with
```

Comment:
It might be good to state that this is essentially a diagonal matrix that can be applied to the spectral coefficients efficiently. While that can be deduced from everything else you say I think being explicit would be good :)